### ISSUE TITLE
obfuscate the code in update.

### ISSUE THREAD

**Issue by FuriosoJack** [YahnisElsts/plugin-update-checker#454]
**Hello, I would like to know if it is possible that the update plugin has a method of obfuscating the code of the plugin that is downloaded in the update?, if it cannot be, well, maybe you could contribute to doing it if you agree to do that practice**

I know that it goes against the GPL of worpdress, however I find that the plugin I made is highly coveted by other developers in the niche in which I find myself, and they have already tried to copy and resell it, because what I now distribute in an obfuscated way, Obviously the plugin works with your code but it is very difficult to understand the code.

I would also like to know what you think about how to combat this problem of having the complete plugin copied, I am a pro of free software and I did not like to go to those extremes but I did not even repeat in deriving my plugin but simply copy it and resell it.

**How do you guys do?**

**Comment by YahnisElsts** [comment]
No, this update checker doesn't have any code obfuscation features. I'm also not particularly interested in adding features like that.

> I would also like to know what you think about how to combat this problem of having the complete plugin copied, I am a pro of free software and I did not like to go to those extremes but I did not even repeat in deriving my plugin but simply copy it and resell it.

Personally, I think there is no good defense against cracking/piracy, so I don't put that much effort into it. I use a pretty basic license verification system without any code obfuscation. My plugins probably get pirated sometimes, but overall I still make a decent income from honest users. I'm not sure how other developers feel about this topic - you might get better advice from someone else.
